std/utf.d

```diff
@@ -3106,11 +3106,28 @@ enum dchar replacementDchar = '\uFFFD';
  * Iterate a range of char, wchar, or dchars by code unit.
  *
  * The purpose is to bypass the special case decoding that
- * $(XREF array,front) does to character arrays.
+ * $(XREF array,front) does for character arrays. As a result, using ranges
+ * with `byCodeUnit` can be `nothrow` while $(XREF array,front) throws when
+ * it encounters invalid Unicode sequences.
+ *
+ * $(RED Warning:) This range does no encoding, so if the string you're passing
+ * to `byCodeUnit` contains things such as Unicode combining characters or
+ * graphemes that take up more than one code unit, then this range will not
+ * encode those values into code points that would produce "correct" Unicode
+ * graphemes.
+ *
```

Comment:
> I'm not sure if I follow. What kind of encoding action could someone expect from byCodeUnit? Of course it doesn't encode (or rather decode) anything to code points; this isn't byCodePoint.

No reason we can't make things explicit for people new to Unicode specifics, e.g. me four months ago.

> by itself byCodeUnit will not break code points or graphemes as that warning seems to say. It just allows the user to (accidentally) break them.

I'll change it to mention "when iterating over" to be more clear